On The Map/Controller/AddLocationViewController.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  AddLocationVC.swift
+//  On The Map
+//
+//  Created by Michal Hus on 2/23/20.
+//  Copyright © 2020 Michal Hus. All rights reserved.
+//
+
+import UIKit
+import MapKit
+import CoreLocation
+
+class AddLocationViewController: UIViewController {
+    
+    var coordinate: CLLocationCoordinate2D = CLLocationCoordinate2D()
+    var activityIndicator: UIActivityIndicatorView = UIActivityIndicatorView()
+    
+    @IBOutlet weak var location: UITextField!
+    @IBOutlet weak var link: UITextField!
+    
+    @IBAction func cancelOut(_ sender: Any) {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func addLocation(_ sender: Any) {
+        
+        activityIndicator.center = self.view.center
+        activityIndicator.hidesWhenStopped = true
+        activityIndicator.style = UIActivityIndicatorView.Style.gray
+        view.addSubview(activityIndicator)
+        activityIndicator.startAnimating()
+        UIApplication.shared.beginIgnoringInteractionEvents()
+        
+        Client.getUserPublicInfo { (success, error) in
+            guard success, error == nil else {
+                self.errorAlertMessage(title: "Fetching User Data Failure", message: error!.localizedDescription)
+                return
+            }
+        }
+        
+        guard let location = location.text, location != ""  else {
+            errorAlertMessage(title: "Location Failure", message: "Please enter a locatin.")
+            return
+        }
+        
+        guard let link = link.text, link != "" else {
+            errorAlertMessage(title: "Link Failure", message: "Please enter a link.")
+            return
+        }
+        
+        let geocoder = CLGeocoder()
+        
+        geocoder.geocodeAddressString(location) { (placemark, error) in
+            self.processResponse(withPlacemarks: placemark, error: error)
```

Comment:
I notice this is the only place you call the function that calls `endIgnoringInteractionEvents()`so presumably if you got an error and returned earlier than this, you will never stop ignoring interaction events and your app will become unresponsive until the user force quits it.

Reply:
That makes sense, need to add `endIgnoringInteractionEvents()` in error handling to stop that from happening.

i.e.

Reply:
note: sometimes you can use defer in swift to do some cleanup when you exit scope like:
```
defer {
    endIgnoringInteractionEvents()
{
```

In your case I think you would want to put the defer inside the completion handler, because you want to reenable interaction after all the tasks have completed.